test.sh

```diff
@@ -2,4 +2,9 @@
 
 GOPATH="$(pwd -P)/../../../.."
 
-./gotest
+TESTDIRS=$(find . -name "*_test.go" -and ! -path "./src/*" | xargs dirname | sort | uniq)
+
+for i in $TESTDIRS; do
+    echo Testing in $i:
+    (cd $i; go test -v)
```

Comment:
You should add the -race flag (https://golang.org/doc/articles/race_detector.html), to detect goroutines concurrent access... And then protect accesses to the Store.client concurrent accesses. See platform access to store.session protected by a mutex here : 
https://github.com/mdblp/platform/pull/64/commits/858a05825fc1ff73b8fb8c7261c6e58fd41c6b0d

```suggestion
    (cd $i; go test -race  -v)
```